### ISSUE TITLE
Searching transactional index missing results

### ISSUE THREAD

**Issue by dennishendriksen** [molgenis/molgenis#4478]
Reproduce:
- Add the test case below to ```ElasticsearchServiceIntegrationTest.java``` and execute the test
```java
@Test
public void transactionalAddAndTransactionAddAndFindAll()
{
	// entity in existing index
	Entity entity0 = new DefaultEntity(entityMeta, dataService);
	entity0.set(idAttrName, "0");
	entity0.set(labelAttrName, "label0");

	elasticsearchService.index(entity0, entityMeta, IndexingMode.ADD);
	elasticsearchService.refresh(entityMeta);
	assertEquals(elasticsearchService.count(entityMeta), 1l);

	String transactionId = "transaction0";
	TransactionSynchronizationManager.bindResource(TRANSACTION_ID_RESOURCE_NAME, transactionId);
	try
	{
		elasticsearchService.transactionStarted(transactionId);

		Entity entity1 = new DefaultEntity(entityMeta, dataService);
		entity1.set(idAttrName, "1");
		entity1.set(labelAttrName, "label1");

		elasticsearchService.index(entity1, entityMeta, IndexingMode.ADD);
		elasticsearchService.refresh(entityMeta);
		assertEquals(elasticsearchService.count(entityMeta), 2l);

		List<Entity> entities = elasticsearchService.searchAsStream(new QueryImpl(), entityMeta).collect(toList());
		assertEquals(entities.size(), 2);
		elasticsearchService.commitTransaction(transactionId);
	}
	finally
	{
		TransactionSynchronizationManager.unbindResource(TRANSACTION_ID_RESOURCE_NAME);
	}
	assertEquals(elasticsearchService.count(entityMeta), 2l);
}
```

Expected:
- Test passes

Actual:
- ```searchAsStream``` returns a list containing 1 instead of 2 entities

This causes for example EMX imports to fail when adding attributes. 

### Note: Please enable testIt test case in org.molgenis.integrationtest.data.elasticsearch.ElasticsearchMetaDataIT when fixing this issue

**Comment by Mark-de-Haan** [comment]
Reverted the workaround as it broke the setting of group permissions.

**Comment by dennishendriksen** [comment]
Fixed in 2.0-RC where the index is updated only after transactions commit.
